src/main/java/net/fabricmc/loom/task/service/MixinMappingsService.java

```diff
@@ -47,13 +48,13 @@ public static File getMixinMappingFile(Project project, SourceSet sourceSet) {
 		final LoomGradleExtension extension = LoomGradleExtension.get(project);
 		File mixinMapping = new File(extension.getFiles().getProjectBuildCache(), "mixin-map-" + extension.getMappingsProvider().mappingsIdentifier() + "." + sourceSet.getName() + ".tiny");
 
-		getService(SharedServiceManager.get(project)).mixinMappings.add(mixinMapping);
+		getService(SharedServiceManager.get(project), extension.getMinecraftProvider()).mixinMappings.add(mixinMapping);
 
 		return mixinMapping;
 	}
 
-	static synchronized MixinMappingsService getService(SharedServiceManager sharedServiceManager) {
-		return sharedServiceManager.getOrCreateService("MixinMappings", () -> new MixinMappingsService(sharedServiceManager));
+	static synchronized MixinMappingsService getService(SharedServiceManager sharedServiceManager, MinecraftProvider minecraftProvider) {
+		return sharedServiceManager.getOrCreateService("MixinMappings-" + minecraftProvider.minecraftVersion(), () -> new MixinMappingsService(sharedServiceManager));
```

Comment:
I think this is along the correct lines, however I think it should be using `extension.getMappingsProvider().mappingsIdentifier()` as used in a few lines above insead of the minecraft version.